datafusion/expr/src/expr.rs

```diff
@@ -1198,12 +1201,21 @@ impl fmt::Display for Expr {
                     write!(f, " NULLS LAST")
                 }
             }
-            Expr::ScalarFunction(func) => {
-                fmt_function(f, &func.fun.to_string(), false, &func.args, true)
-            }
-            Expr::ScalarUDF(ScalarUDF { fun, args }) => {
-                fmt_function(f, fun.name(), false, args, true)
-            }
+            Expr::ScalarFunction(func_expr) => match &func_expr.func_def {
```

Comment:
We could add a function like 

```rust
impl ScalarFunctionDefinition {
  pub fn name(&self) -> &str {
  ..
  }
}
```

And could reduce the duplication in this method more